src/main/java/Duke.java

```diff
@@ -1,10 +1,289 @@
+                num += c;
+            }
+        }
+
+        int index = Integer.parseInt(num);
+        Task removedTask = tasks.get(index - 1);
+        tasks.remove(index - 1);
+        System.out.println(line
+                + "Noted. I've removed this task: \n"
+                + removedTask.toString()
+                + "\nNow you have " + tasks.size()
+                + " tasks in the list.\n" + line);
+    }
+
+    public static final void clearList(ArrayList<Task> tasks) {
+        while (tasks.size() > 0) {
+            tasks.remove(0);
+        }
+        
+        System.out.println(line
+                + "The list has been cleared. Now the list is empty.\n"
+                + line);
+    }
+
+    public static final ArrayList<Task> getDatafromFile(String fileName) {
+        ArrayList<Task> tasks = new ArrayList<Task>();
+        String path = "C:\\Users\\XUEY0013\\Documents\\ip\\src\\main\\" + fileName+ ".txt";
+        File file = new File(path);
+        if (!file.exists()) {
+            return new ArrayList<Task> ();
+        }
+        BufferedReader reader = null;
+        try {
+            FileInputStream fileInputStream = new FileInputStream(path);
+            InputStreamReader inputStreamReader = new InputStreamReader(fileInputStream, "UTF-8");
+            reader = new BufferedReader(inputStreamReader);
+            String tempString = "";
+            Task task = new Task("null");
+
+            while ((tempString = reader.readLine()) != null) {
+                if (tempString.charAt(TYPE_POS) == 'T') {
+                    task = (new Todo(tempString.substring(TASK_POS)));
+                } else if (tempString.charAt(TYPE_POS) == 'E') {
+                    String event = tempString.substring(TASK_POS,
+                            tempString.indexOf("(") - 1);
+                    String at = tempString.substring(tempString.indexOf(":") + 2,
+                            tempString.indexOf(")"));
+                    task = (new Event(event, at));
+                } else if (tempString.charAt(TYPE_POS) == 'D') {
+                    String deadline = tempString.substring(TASK_POS,
+                            tempString.indexOf("(") - 1);
+                    String by = tempString.substring(tempString.indexOf(":") + 2,
+                            tempString.indexOf(")"));
+                    task = (new Deadline(deadline, by));
+                }
+
+                if (tempString.charAt(7) == 'X') {
+                    task.markedAsDone();
+                }
+
+                tasks.add(task);
+            }
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        } finally {
+            if (reader != null) {
+                try {
+                    reader.close();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+        return tasks;
+    }
```

Comment:
Having such long functions can be hard to read. Please consider abstracting out repeated code in line with the DRY (Don't Repeat Yourself) principle. Also add comments to improve the readability of your code